### PR TITLE
[cloud_firestore] add support for MetadataChanges

### PR DESCRIPTION
**Pull request by long1eu**
## Related Issues
https://github.com/flutter/flutter/issues/16432

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by long1eu** [comment]
It's a bit strange. I'm running `pub global activate flutter_plugin_tools && pub global run flutter_plugin_tools format` but I get a different result than the one from cirrus. 

The difference was on the flutter version, I used the stable channel.

File: packages/cloud_firestore/example/lib/main.dart [hunk 1]
```diff
@@ -42,7 +42,7 @@ class MessageList extends StatelessWidget {
           itemBuilder: (_, int index) {
             final DocumentSnapshot document = snapshot.data.documents[index];
             return ListTile(
-              title: Text(document['message'] ?? '<No message retrieved>'),
+              title: Text('${document['message']}' ?? '<No message retrieved>'),
```
Comment (collinjackson) [inline packages/cloud_firestore/example/lib/main.dart]: I don't think is quite what you want, we want to show '<No message retrieved>' if the message is `null`.

**Comment by collinjackson** [comment]
Thanks for the PR! LGTM

**Comment by long1eu** [comment]
Do you want me to fix this? Also please make sure to assign me 2 issues each week. I prefer the ones related to plugins. :D

**Comment by collinjackson** [comment]
I think I fixed the test. Do you have any feedback/changes you want to make before I merge this?

Here's an auth issue you might consider taking on: https://github.com/flutter/flutter/issues/35974

Generally we try prioritize PRs that are associated with issues that have a lot of thumbs. Feel free to reach out to me on Gitter or via my email jackson@google.com for advice on issue prioritization.

**Comment by long1eu** [comment]
LGTM, also I can take https://github.com/flutter/flutter/issues/35974

**Comment by collinjackson** [comment]
> LGTM, also I can take [flutter/flutter#35974](https://github.com/flutter/flutter/issues/35974)

Great, thank you. I put a few other issue ideas in the list above.

**Pull request merged**